autogen/main/cluster.tf.tmpl

```diff
@@ -544,8 +544,8 @@ resource "google_container_cluster" "primary" {
     dynamic "additional_ip_ranges_config" {
       for_each = var.additional_ip_ranges_config
       content {
-        subnetwork           = var.additional_ip_ranges_config.subnetwork
-        pod_ipv4_range_names = var.additional_ip_ranges_config.pod_ipv4_range_names
+        subnetwork           = var.additional_ip_ranges_config.value.subnetwork
+        pod_ipv4_range_names = var.additional_ip_ranges_config.value.pod_ipv4_range_names
```

Comment:
Shouldn't this be like this instead?
```suggestion
        subnetwork           = additional_ip_ranges_config.value.subnetwork
        pod_ipv4_range_names = additional_ip_ranges_config.value.pod_ipv4_range_names
```

Reply:
Based on the docs at https://developer.hashicorp.com/terraform/language/expressions/dynamic-blocks

Reply:
+1

@DrFaust92 - Can you please add a usage to the existing [node_pool](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/main/examples/node_pool/main.tf#L50) example and [test data](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/main/test/integration/node_pool/testdata/TestNodePool.json) so we have sufficient test coverage.  Thanks!

Reply:
Ok, I think i added the changes and test. can we trigger a run?